boards/arduino/nano_matter/doc/index.rst

```diff
@@ -0,0 +1,116 @@
+.. zephyr:board:: arduino_nano_matter
+
+Overview
+********
+
+The Nano Matter merges the well-known Arduino way of making complex technology more accessible with the powerful MGM240S from Silicon Labs, to bring Matter closer to the maker world, in one of the smallest form factors in the market.
```

Comment:
This is the "source file" of the documentation, so I suggest to apply 80 columns rule.